bin/focus-send

```diff
@@ -17,29 +17,47 @@ set -e
 
 OS=$(uname -s)
 
+# igncr absorbs CR from Focus CRLF line endings
+# -echo is needed because raw doesn't turn it off on Linux
+STTY_ARGS="9600 raw igncr -echo"
+
 case ${OS} in
     Linux)
         DEVICE="${DEVICE:-/dev/ttyACM0}"
-        stty -F "${DEVICE}" 9600 raw -echo
         ;;
     Darwin)
+        # bash on macOS has a bug that randomly drops serial input
+        if [ -n "$BASH_VERSION" ] && [ -x /bin/dash ]; then
+            # Prevent loop in case someone exported it
+            export -n BASH_VERSION
+            exec /bin/dash "$0" "$@"
+        fi
         DEVICE="${DEVICE:-/dev/cu.usbmodemCkbio01E}"
-        stty -f "${DEVICE}" 9600 raw -echo
         ;;
     *)
         echo "Error Unknown OS : ${OS}" >&2
         exit 1
         ;;
 esac
 
+# Redirect prior to running stty, because macOS sometimes resets termios
+# state upon last close of a terminal device.
+exec < "${DEVICE}"
+# shellcheck disable=SC2086 # intentional word splitting
+stty $STTY_ARGS
 
-exec 3<"${DEVICE}"
-echo "$@" >"${DEVICE}"
+wait_dot () {
```

Comment:
Not entirely sold on the name, because it doesn't just wait for a dot, but echoes the reply as well. Maybe `read_reply` instead?